Added ability to set local layout to null

EthanRBrown opened the pull request:
Currently, when `defaultLayout` is set, setting `layout` to `null` locally does not result in a view rendering without a layout, which is a desirable feature.

All tests passing; no new tests added.

dominicbarnes commented:
I'd prefer to see a `typeof` check here. (ie: `typeof locals.layout === "undefined"`)

Adding a test would be preferred as well.

EthanRBrown commented:
Switched to using `typeof` check (I generally reserve `typeof` checks for local/automatic variables, but for consistency, I understand the reasons for this), and added test, per request.

dominicbarnes commented:
Beautiful! Thanks :)

dominicbarnes commented:
Released as part of `0.5.4`
